zerver/tests/test_reactions.py

```diff
@@ -265,6 +256,37 @@ def test_remove_nonexisting_reaction(self):
                                     **self.api_auth(reaction_sender))
         self.assert_json_error(second, "Reaction does not exist")
 
+    def test_remove_existing_reaction_with_renamed_emoji(self):
+        # type: () -> None
+        """
+        Removes an old existing reaction but the name of emoji got changed during
+        various emoji infra changes.
+        """
+        sender = self.example_email("hamlet")
+        result = self.client_put('/api/v1/messages/1/emoji_reactions/smile',
+                                 **self.api_auth(sender))
+        self.assert_json_success(result)
+
+        with mock.patch('zerver.lib.emoji.name_to_codepoint', name_to_codepoint={}):
+            result = self.client_delete('/api/v1/messages/1/emoji_reactions/smile',
+                                        **self.api_auth(sender))
+            self.assert_json_success(result)
+
+    def test_remove_existing_reaction_with_deactivated_realm_emoji(self):
+        # type: () -> None
+        """
+        Removes an old existing reaction but the realm emoji used there has been deactivated.
+        """
+        sender = self.example_email("hamlet")
+        result = self.client_put('/api/v1/messages/1/emoji_reactions/green_tick',
+                                 **self.api_auth(sender))
+        self.assert_json_success(result)
+
+        with mock.patch('zerver.models.Realm.get_emoji',
+                        return_value={'green_tick': {'deactivated': True}}):
+            result = self.client_delete('/api/v1/messages/1/emoji_reactions/green_tick',
+                                        **self.api_auth(sender))
+            self.assert_json_success(result)
```

Comment:
Can you just deactivate the emoji instead of using the `with` statement here?  I'd much prefer something end-to-end to doing mocking, since the test as written is sorta fragile.

Reply:
When I looked at again it, I also wondered why I did it this way. Sorry about this, updated.